docs/limitations.md

```diff
@@ -26,8 +26,17 @@ The [conditional execution](https://www.commonwl.org/v1.2/Workflow.html#Conditio
 
 ## Dealing with type incompatibility, e.g. mismatch of EDAM format
 
+Data types for tool inputs, and outputs, can (and should) be specified using the `format` field, as described in the [File Formats](https://www.commonwl.org/user_guide/16-file-formats/) tutorial. Doing so helps to ensure data consistency throughout a workflow, by checking type compatibility when data produced by one tool is used by another tool, as well as guiding users in what input data to provide (and what output data to expect). 
+
+It is recommended, where possible, to use a format definition taken from a widely-used ontological resource, such as [EDAM](http://edamontology.org). Doing so will enable users to reference the formats that your workflow expects, so that they can ensure they match these.
+
+Note, however, that the cwltool currently compares the format string only, it does not check the file for conformance to that format definition. It is up to you to make sure that you use the correct definition for your file format. If a format definition for your files is not available in the existing ontological databases, then you can set your own format string - however it is recommended that you do this in your own namespace, so it is clear it is not part of any existing database. It is also possible to request the addition of your format to a database, e.g. by raising an [issue for EDAM on github](https://github.com/edamontology/edamontology/issues).
+
 ## Managing file names and file storage
 
+CWL does ensure separation of working space for each step within a workflow, however output files cannot be renamed as they are gathered from steps, and the requested outputs are returned to the base directory from which the workflow is run. This can cause issues when using `scatter` to parallise a workflow, as outputs can potentially overwrite each other. It also can result in a messy workflow directory, if a lot of outputs are being returned.
+
+One solution for this is to add a final step to your workflow that gathers together all desired outputs into a single directory. This can most easily be performed using an `ExpressionTool` to create a uniquely named directory for each scatter instance containing the required files. This will need the  `InlineJavascriptRequirement` requirement though, and so will mean your workflow is less portable than it would otherwise be.
```

Comment:
>  and so will mean your workflow is less portable than it would otherwise be.

Basically all CWL engines support `InlineJavascriptRequirement`, though it is technically an optional part (like supporting `Workflow`s at all :-)

So I would appreciate it if this statement was removed

Reply:
Yep - happy to modify/remove this statement. I was thinking of the requirement of installing a java engine to be able to support these tools, which I had issues with when first learning CWL. So I could perhaps reword this too:
"This will need the `InlineJavascriptRequirement` requirement though, and to support using this you will need to install java (using, for example, the conda-forge `nodejs` package) or docker (enabling CWL to access java support via the `openjdk` docker image)."

Reply:
"This will need the InlineJavascriptRequirement requirement though, and to support using this with `cwltool` or `toil-cwl-runner` you will need to install the Node javascript engine (using, for example, the conda-forge `nodejs` package) or docker (enabling CWL to access javascript support via the nodejs docker image)."

Reply:
I've opened a new [PR](https://github.com/bioexcel/cwl-best-practice-guide/pull/29) for this correction - could you have a look at that and let me know if it looks okay?